server/auth_function/test/lamda_function_test.py

```diff
@@ -5,20 +5,34 @@
 
 import pytest
 from psycopg2 import sql
+from constant import TEST_ROLE_NAME
 
 modulePath = os.path.join(os.path.dirname(__file__), "..")
 sys.path.append(modulePath)
 import lambda_function  # noqa
+from util import (
+    format_user_properties,
+    initial_user_without_guid_or_cognito_id,
+    initial_user,
+    create_user_role_xref_record,
+    create_test_fam_cognito_client,
+)
 
 LOGGER = logging.getLogger(__name__)
-TEST_ROLE_NAME = "EXPECTED"
-
 
-def test_create_user_if_not_found(
-    db_pg_transaction, cognito_event, cognito_context, test_user_properties
-):
 
+@pytest.mark.parametrize(
+    "cognito_event",
+    [
+        "login_event.json",
+        "login_event_bceid.json",
+        "login_event_bcsc.json",
+    ],
+    indirect=["cognito_event"],
+)
+def test_create_user_if_not_found(db_pg_transaction, cognito_event, cognito_context):
```

Comment:
parametrize the cognito_event fixture method, so by given different json file, it will return different data to run the test